buildtools/yarn/yarn.go

```diff
@@ -0,0 +1,45 @@
+package yarn
+
+import (
+	"os"
+
+	"github.com/apex/log"
+	"github.com/fossas/fossa-cli/exec"
+)
+
+type YarnTool interface {
+	// TODO: implement yarn list --production --frozen-lockfile. Skipping for now because the implementation relies 100% on yarn.lock
+	// List(dir string) error
+	Install(dir string) error
+	// Exists returns true if yarn is available
+	Exists() bool
+}
+
+type SystemYarn struct {
+	Cmd string
+}
+
+func New() YarnTool {
+	yarnCmd, _, yarnErr := exec.Which("-v", os.Getenv("FOSSA_NPM_CMD"), "yarn")
+	if yarnErr != nil {
+		log.Warnf("Could not find Yarn %s", yarnErr.Error())
```

Comment:
We should return this error instead of logging.

In general, errors should be propagated outwards as far as possible. Fewer side effects makes code easier to reason about.